hyperon_das_atomdb/database.py

```diff
@@ -273,22 +354,109 @@ def clear_database(self) -> None:
+                    'type': 'Reactome',
+                    'name': 'Reactome:R-HSA-164843',
+                }
+            >>> db.add_node(node_params)
         """
         ...  # pragma no cover
 
-    def add_link(self, link_type: str, targets: List[str]) -> None:
+    @abstractmethod
+    def add_link(self, link_params: Dict[str, Any], toplevel: bool = True) -> Dict[str, Any]:
         """
-        Add a link with the specified type and targets to the database.
+        Adds a link to the database.
+
+        This method allows to add a link to the database with the specified
+            link parameters.
+        A link must have a 'type' and 'targets' field
+            in the link_params dictionary.
 
         Args:
-            link_type (str): The link type.
-            targets (List[str]): A list of link target identifiers.
+            link_params (Dict[str, Any]): A dictionary containing
+                link parameters.
+                It should have the following keys:
+                - 'type': The type of the link.
+                - 'targets': A list of target elements.
+            toplevel: boolean flag to indicate toplevel links
+                i.e. links which are not nested inside other links.
+
+        Returns:
+            Dict[str, Any]: The information about the added link,
+                including its unique key and other details.
+
+        Raises:
+            AddLinkException: If the 'type' or 'targets' fields
+                are missing in link_params.
+
+        Note:
+            This method supports recursion when a target element
+                itself contains links.
+            It calculates a unique key for the link based on
+                its type and targets.
+            If a link with the same key already exists,
+                it just returns the link.
+
+        Example:
+            To add a link, use this method like this:
+            >>> link_params = {
+                    'type': 'Evaluation',
+                    'targets': [
+                        {
+                            'type': 'Predicate',
+                            'name': 'Predicate:has_name'
+                        },
+                        {
+                            'type': 'Set',
+                            'targets': [
+                                {
+                                    'type': 'Reactome',
+                                    'name': 'Reactome:R-HSA-164843',
+                                },
+                                {
+                                    'type': 'Concept',
+                                    'name': 'Concept:2-LTR circle formation',
+                                },
+                            ],
+                        },
+                    ],
+                }
+            >>> db.add_link(link_params)
+        """
+        ...  # pragma no cover
+
+    @abstractmethod
+    def update_index(self, handles: Any) -> None:
+        """
```

Comment:
So it would be interesting to add documentation here

Reply:
Yes. It's not done yet because the concrete method is not implemented yet in redis_mongo. So CHanges in the parameters/return may happen.